### ISSUE TITLE
Node can't sync if it is stopped mid sync and then restarted

### ISSUE THREAD

**Issue by divagant-martian** [sigp/lighthouse#1635]
## Description
As the title says. I tried with a fresh node, no db, no changes in the code. Let it start syncing, stop it, restart it and it won't sync.
What I see:
batches fail with unknown parent until the chain is removed
another chain will be started from the same finalized point and the same will happen again and again.

## Version
`master, tag: v0.2.10`

## Present Behaviour
Can't sync if stopped mid sync and restarted

## Expected Behaviour
Be able to sync

## Steps to resolve
no idea

**Comment by paulhauner** [comment]
Here are some logs I got from @divagant-martian out-of-band.

## Logs

```
Sep 19 23:55:30 localhost lighthouse[219520]: Sep 19 23:55:30.001
DEBG Sending Status Request
head_slot: 58432, 
head_root: 0x1058…c871, 
finalized_epoch: Epoch(1824), 
finalized_root: 0x4f2f7a8acbf16e4372585b6d29191025177467dea8472beb97322d331d4c79d7, 
fork_digest: [231, 167, 93, 90], 
peer: 16Uiu2HAky1bBKhykDhiPyngC1bE4Lpt41XQ7NkJfKyHngVxCpm7V
robust batch failure: 
```

```
Sep 19 23:57:56 localhost lighthouse[219520]: Sep 19 23:57:56.350
DEBG Batch processing failed
service: sync, imported_blocks: 0, 
error: Block has an unknown parent: 0x4f2f…79d7, 
last_block_slot: 58432, 
first_block_slot: 58370, 
batch_epoch: 1824
but the optimistic also fails with the root of our head 
```
```
Sep 19 23:56:45 localhost lighthouse[219520]: Sep 19 23:56:45.385
DEBG Batch processing failed
service: sync, imported_blocks: 0, 
error: Block has an unknown parent: 0x1058…c871, 
last_block_slot: 58495,
first_block_slot: 58433, 
batch_epoch: 1826
```

**Comment by AgeManning** [comment]
Just to make things clearer. 

`beacon_chain.head_info().finalized_checkpoint` Results with a known hash, i.e `0x4f2f7a..` in this case. Then when processing a block that has this hash as it's parent, the block processor returns parent unknown `0x4f2fza..` which is clearly known.

**Comment by paulhauner** [comment]
I'm digging into this now, I believe I'm reproducing the bug.

**Comment by AgeManning** [comment]
Closed in #1616
